modules/dhcp_native_ms/dhcp_native_ms_main.rb

```diff
@@ -91,28 +91,28 @@ def find_or_create_vendor_name vendor, exception
     private :find_or_create_vendor_name
 
     def find_subnet_dhcp_records(subnet)
-      cmd = "scope #{subnet.network} show reservedip"
+      cmd = "scope #{subnet.network} show client 1"
       msg = "Enumerated hosts on #{subnet.network}"
 
       to_return = []
       # Extract the data
       execute(cmd, msg).each do |line|
-        #     172.29.216.6      -    00-a0-e7-21-41-00-
-        if line =~ /^\s+([\w\.]+)\s+-\s+([-a-f\d]+)/
-          ip  = $1
+        #192.168.205.4    - 255.255.255.128- 00-1e-68-65-55-f8   -4/23/2013 6:41:21 PM   -D-
+        #192.168.205.5    - 255.255.255.128-00-1b-24-93-35-09   - NEVER EXPIRES        -U-  host.brs.company.com
+        if (match = line.match(/\A([\d\.]+)\s*-\s*[\d\.]+\s*- ?([-a-f\d]+)\s*-\s*([^-]+?)\s*-(\w)-\s*(.*)/))
+          ip, mac, expire, kind, name  = match[1,5]
           next unless subnet.include?(ip)
-          mac = $2.gsub(/-/,":").match(/^(.*?).$/)[1]
+          # Some mac addresses appear to be more than 6 bytes!
+          mac = mac[0,17].gsub!(/-/,":")
           begin
-            opts = {:subnet => subnet, :ip => ip, :mac => mac}
-            opts.merge!(loadRecordOptions(opts))
+            opts = {:subnet => subnet, :ip => ip, :mac => mac, :name => name}
             logger.debug opts.inspect
-            if opts.include? :hostname
-              to_return << Proxy::DHCP::Reservation.new(opts.merge(:deleteable => true))
+            if kind == 'D' && expire !~ /^INACTIVE|^NEVER/
```

Comment:
You may well be right on this point. This logic works for my organisation but this may not be generally true.
I will update the push to remove the "D" clause as just using the INACTIVE|NEVER test would return the same host list.